clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -84,6 +85,9 @@ private enum SubscriptionType {
     /* the pattern user has requested */
     private Pattern subscribedPattern;
 
+    /* we should rename this to something more specific */
+    private SubscriptionPattern subscriptionPattern;
```

Comment:
Could you please also reset this field in `unsubscribe()` and add it to the return value of `toString()`?

Reply:
Will do